summaries/Evolutionary-testing-of-classes-summary.md

```diff
@@ -0,0 +1,82 @@
+
+### Chromosomes
+Chromosomes include a number of input values to be used as parameters of methods and constructors that belong to a certain class.
+The specification of chromosomes thus must include the specification of the methods and the proper sequence of parameters.
+
+### Mutation operators
+By randomly selecting mutation operators, chromosomes are being changed by the genetic algorithm.
+#### Mutation of input value
+Randomly generated value replaces another with the same type.
+#### Constructor change
+One of the constructors is randomly replaced.
+#### Insertion of method invocation
+New methods are generated with generated input values
+#### Removal of method invocation
+Randomly chosen methods are being removed.
+#### One-point crossover
+During a union of chromosomes at a randomly selected line, the confliction variables are being renamed, constructors are added or removed, methods are added or removed.
+
+### Input generators
+Input generators are defined in the declaration of method signatures. 
+Default input generator are use if only type names are declared.
+The following types for generated rules are applied:
+- Integer and real numbers (only in the interval from 0 to 100)
+- Booleans (50% probability for true, and 50% for false)
+- Strings (a-zA-Z0-9) 
+
+##The Tool eToc
+The Evolutionary Testing of Classes tool (eToc) uses the genetic algorithm for test case generation in the terms of the Java language.
+It consists of the following components:
+
+###Branch instrumentor
+Branch instrumentor generates instrumented code where each control flow branch is individually described and traced in the process of execution. 
+
+### Chromosome former
+Produces new chromosomes and modifies existing chromosomes according to method signatures. 
+
+### Test case generator 
+Within the inner loop of the genetic algorithm every test case identified by a chromosome is implemented.
+
+#### Test case executor
+After execution of test cases, a summary containing information of successfully passed test cases, as well as failed ones and not proper assertions, is being generated.
+
+## Experimental Results
+Experimental results have been generated by using eToc tool.
+The unit testing have been applied to classes existing in the standard Java library.
+
+###The procedure
+The Classes Under Test have been implemented using *Branch instrumentor*.
+After that *Test case generator* generates Junit test cases for each Class Under Test. 
+
+###Classes Under Test
+The classes that have been tested with the given tool belong to standard Java library and SDK version 1.4.0.
+
+###Results
+The outcome of the generated tests that use chromosomes repeat the theoretical assumption that the branch coverage conducted by the experiment equals to 100%.
+On the other hand, the effectiveness of automatically generated tests to detect faults is to some extent high but not completely satisfying. 
```

Comment:
Maybe add a reason why it is not satisfying? This might be relevant for the overall summary.